### PR TITLE
replace explicit PDG id with enum

### PR DESCRIPTION
**Pull request by rlcee**
validated with ceSimReco.  This was a lot of hand edits, so it might be a good idea for someone to double-check the signs.

### CODE REVIEW

**Comment by rlcee** [comment]
Reference #292

**Comment by kutschke** [comment]
I have invited a number of people to review this PR.  In a few minutes I will post a list of files for each to review.  Mostly I want you to be aware that this change has been made to your code and to conform to this pattern in future work.  I will review the files not assigned to anyone else.  I will also review in place of a few people that are no longer on Mu2e or whom I know to be on vacation today.

**Comment by kutschke** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
Here are the review assignments.  please approve when you are done.

andrei
       Analyses/src/pbars1hist_module.cc
       EventGenerator/src/ExtMonFNALBoxGenerator_module.cc
       Filters/src/EMFBoxHitsFilter_module.cc
       Sources/src/ExtMonFNALMARSUtils.cc
       Sources/src/FromExtMonFNALMARSFile_source.cc
gaini
       Analyses/src/ReadCaloDigi_module.cc
       Trigger/src/ReadTriggerInfo_module.cc
pasha
       CalPatRec/src/DeltaFinder2Diag_tool.cc
       CalPatRec/src/DeltaFinderAna_module.cc
       CalPatRec/src/DeltaFinderDiag_tool.cc
ralf
       Analyses/src/CosmicAnalysis_module.cc
       CRVAnalysis/src/CRVAnalysis.cc
       EventDisplay/src/TrackColorSelector.cc

rob    Analyses/src/CORSIKAGenPlots_module.cc
       Analyses/src/CRYGenPlots_module.cc
       Analyses/src/GeomVis_module.cc
       Analyses/src/PbarAnalysis2_module.cc
       EventGenerator/src/AntiProtonGun_module.cc
       Print/src/SimParticlePrinter.cc
       TrkDiag/src/BkgDiag_module.cc
       Validation/src/ValId.cc
       Validation/src/ValSimParticle.cc
       Validation/src/ValTrackSummary.cc
       TrkExt/src/TrkExtDiag.cc

sophie
       CosmicReco/src/CosmicMCRecoDiff_module.cc
       CosmicReco/src/CosmicTrackDiag_module.cc

**Review by ehrlich-uva — APPROVED** [review]

**Review by kutschke — CHANGES_REQUESTED** [review]

**Comment by kutschke** [comment]
@rlc One of the merges that I did earlier today created a conflict.  Please resolve it.  Thanks.

File: Analyses/src/PbarAnalysis2_module.cc [hunk 1]
```diff
@@ -528,8 +528,8 @@ namespace mu2e {
             // very unfortunately implementation dependent: how many stages
             //
             // set these now, write them and initial proton momentum vector later
-            if (p.id().asInt() == 2000001 && p.pdgId() == -2212){
-              //              if (p.pdgId() == -2212){
+            if (p.id().asInt() == 2000001 && p.pdgId() == PDGCode::anti_proton){
```
[the remark this answers is not in the document]
Comment (kutschke) [reply Analyses/src/PbarAnalysis2_module.cc:531]: I opened the issue: see issue #845 .  There is nothing for you to do here.

**Review by sophiemiddleton — APPROVED** [review]
The files you pointed me too look good

**Review by kutschke — APPROVED** [review]

File: Sources/src/ExtMonFNALMARSUtils.cc [hunk 2]
```diff
@@ -75,46 +75,46 @@ namespace mu2e {
...
+        table[2] = PDGCode::n0; // neutron
+        table[3] = PDGCode::pi_plus; // pi+
+        table[4] = PDGCode::pi_minus; // pi-
+        table[5] = PDGCode::K_plus; // K+
+        table[6] = PDGCode::K_minus; // K-
+        table[7] = PDGCode::mu_plus; // mu+
+        table[8] = PDGCode::mu_minus; // mu-
+        table[9] = PDGCode::gamma; // gamma
+        table[10] = PDGCode::e_plus; // e-
+        table[11] = PDGCode::e_minus; // e+
```
Comment (gaponenko) [inline Sources/src/ExtMonFNALMARSUtils.cc:88]: In ExtMonFNALMarsUtils.cc  e_plus and e_minus are swapped, please fix

File: Sources/src/FromExtMonFNALMARSFile_source.cc [hunk 3]
```diff
@@ -118,46 +118,46 @@ namespace mu2e {
...
+      table[2] = PDGCode::n0; // neutron
+      table[3] = PDGCode::pi_plus; // pi+
+      table[4] = PDGCode::pi_minus; // pi-
+      table[5] = PDGCode::K_plus; // K+
+      table[6] = PDGCode::K_minus; // K-
+      table[7] = PDGCode::mu_plus; // mu+
+      table[8] = PDGCode::mu_minus; // mu-
+      table[9] = PDGCode::gamma; // gamma
+      table[10] = PDGCode::e_plus; // e-
+      table[11] = PDGCode::e_minus; // e+
```
Comment (gaponenko) [inline Sources/src/FromExtMonFNALMARSFile_source.cc:131]: Another case of e+/e- swap.   Perhaps this code should be factored out (as another PR).

**Review by gaponenko — CHANGES_REQUESTED** [review]
The changes are straightforward and mostly OK for the files assigned to me,  but I spotted two typos (e+/e- swaps).  Please fix.
Andrei

File: Sources/src/ExtMonFNALMARSUtils.cc (hunk 2 above)
Comment (rlcee) [reply Sources/src/ExtMonFNALMARSUtils.cc:88]: From PDGCode.hh
        e_minus = 11 ,
        e_plus = -11 ,
which says the code edit was correct, but the comments are wrong, please double-double-check, I might still be reading something wrong..  If you agree, I'll change the comment.  The other question is whether the MARS translation (table indices) is correct.

Comment (gaponenko) [reply Sources/src/ExtMonFNALMARSUtils.cc:88]: Hi Ray,   the comments are correct (at least consistent with MARS documentation).   The old code was wrong, and this became explicit with the code improvements you made in this PR.    Probably this PR should be merged as is, and we can fix the bug in a separate one (which might also handle the code duplication).

**Review by gaponenko — APPROVED** [review]

File: Sources/src/ExtMonFNALMARSUtils.cc (hunk 2 above)
Comment (rlcee) [reply Sources/src/ExtMonFNALMARSUtils.cc:88]: I am looking at fixing the duplicate code in the CRY and CORSIKA modules in this PR.  I could also turn these two methods into a either a free-standing function (with a static member) or a little helper class with an initialized member - let me know which you prefer.

**Comment by kutschke** [comment]
@pavel1murat and @gianipez I am crowd sourcing the review of this PR. There are a few files assigned to each of you.   This is straightforward.  Please get to tihs at your earliest convenience.

**Comment by gaponenko** [comment]
Hi Ray,

Utility code for this already exists in  
Sources/inc/ExtMonFNALMARSUtils.hh and .cc, I don't remember why the 
module does not use it.  May be I just forgot to update that module.

Andrei

**Comment by rlcee** [comment]
> Hi Ray, Utility code for this already exists in Sources/inc/ExtMonFNALMARSUtils.hh and .cc, I don't remember why the module does not use it. May be I just forgot to update that module. Andrei

Shall I make this change or do you want to?

**Comment by gaponenko** [comment]
Please go ahead.
Andrei

**Comment by kutschke** [comment]
@FNALbuild run build test

**Comment by kutschke** [comment]
@pavel1murat and @gianipez I am crowd sourcing the review of this PR. There are a few files assigned to each of you. This is straightforward. Please get to tihs at your earliest convenience.  If  I don't here from you, I plan to merge this tomorrow.  In the future please use the enums, not the numeric literals for PDG ids.

**Comment by kutschke** [comment]
I have not heard back from Giani and Pasha.  I will merge this at the first convenient opportunity.

File: Sources/src/ExtMonFNALMARSUtils.cc (hunk 2 above)
Comment (kutschke) [reply Sources/src/ExtMonFNALMARSUtils.cc:88]: @gaponenko and @rlcee can any remaining issues here be addressed in a followup PR?  I would like to merge today or tomorrow.

Comment (kutschke) [reply Sources/src/ExtMonFNALMARSUtils.cc:88]: ooops - I just saw the comment a little later on the page that says that a follow PR will work.  No need to reply.

**Pull request merged**